### PR TITLE
Update the minimum supported Rust version to 1.29.

### PR DESCRIPTION
**Pull request by sunfishcode**
We no longer need the Ubuntu LTS restriction, so now the only only
constraint I'm aware of is Firefox's policy. Fortunately, that tracks
the latest stable delayed by only two weeks. So this puts is at
Rust 1.29 now.

If anyone is aware of other important constraints that we should follow,
please bring them up!

### CODE REVIEW

File: .travis.yml [hunk 1]
```diff
@@ -2,9 +2,9 @@
 
 language: rust
 rust:
-    - stable
```
Comment (bjorn3) [inline .travis.yml]: Why did you remove stable?

**Comment by bjorn3** [comment]
```
[…]
======  Checking python source files  ======
   === flake8: 3.6.0 (mccabe: 0.6.1, pycodestyle: 2.4.0, pyflakes: 2.0.0) CPython 3.4.3 on Linux ===
./base/instructions.py:1145:-316: W605 invalid escape sequence '\p'
./cdsl/instructions.py:349:-5: W605 invalid escape sequence '\s'
./cdsl/test_ti.py:111:-405: W605 invalid escape sequence '\i'
./cdsl/ti.py:435:-296: W605 invalid escape sequence '\ '
./cdsl/ti.py:435:-29: W605 invalid escape sequence '\ '
./cdsl/typevar.py:124:-28: W605 invalid escape sequence '\i'
./cdsl/xform.py:119:-129: W605 invalid escape sequence '\i'
./semantics/__init__.py:27:-254: W605 invalid escape sequence '\i'
./semantics/__init__.py:27:-71: W605 invalid escape sequence '\i'
[…]
```

File: .travis.yml (hunk 1 above)
Comment (sunfishcode) [reply .travis.yml]: We don't want to immediately update to stable when new versions are released because we're following [Firefox's policy](https://wiki.mozilla.org/Rust_Update_Policy_for_Firefox).

**Comment by sunfishcode** [comment]
A new version of flake8 added some new warnings. These are now fixed.

**Comment by lachlansneff** [comment]
This is great! Once this lands, I'll update preopt to use rustc_apfloat.

**Comment by sunfishcode** [comment]
Ok, let's land this!
